packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -242,6 +239,50 @@ export class DataProxyEngine extends Engine {
     return [host, apiKey]
   }
 
+  private mergeOverriddenDatasources(): Record<string, InlineDatasource> {
+    if (this.config.datasources === undefined) {
+      return this.inlineDatasources
+    }
+
+    const finalDatasources = klona(this.inlineDatasources)
+
+    for (const override of this.config.datasources) {
```

Comment:
That's when I wish we made blaze a utility library for all the packages :crying_cat_face: 😄:
```ts
return { ....inlineDatasources, ....map(config.datasources ?? {}, (d) => ({ url: { value: d.url } })) }
```

Reply:
This way of thinking is still helpful even without the actual utility, I simplified code based on this comment, thank you!